src/java.base/share/classes/sun/net/www/protocol/http/DigestAuthentication.java

```diff
@@ -84,23 +108,23 @@
 
     static {
         @SuppressWarnings("removal")
-        boolean flag = AccessController.doPrivileged(
+        Boolean b = AccessController.doPrivileged(
             new PrivilegedAction<>() {
                 public Boolean run() {
-                    Boolean b = NetProperties.getBoolean(compatPropName);
-                    return (b == null) ? false : b.booleanValue();
+                    return NetProperties.getBoolean(compatPropName);
                 }
             }
         );
-        delimCompatFlag = flag;
+        delimCompatFlag = (b == null) ? false : b.booleanValue();
 
         disabledAlgs.addAll(defDisabledAlgs);
         @SuppressWarnings("removal")
         String algs = AccessController.doPrivileged(
             new PrivilegedAction<>() {
                 public String run() {
                     String s = NetProperties.get(enabledAlgPropName);
-                    return s == null ? "" : s;
+                    return s == null
+                        ? "" : s.replaceAll("\\s", "").toUpperCase();
```

Comment:
Should probably use Local.ROOT to convert to upper case.
It seems to me that the code that takes a String as argument, check for null and returns an empty set, remove spaces, convert it to upper case, splits the string at commas, and create an immutable set from that, could be moved to an auxillary function and called for parsing both the Security property and the System property - since their syntax is identical.

Reply:
good idea